webapp/channels/src/utils/utils.tsx

```diff
@@ -1355,7 +1355,7 @@ function isChannelOrPermalink(link: string) {
 export async function handleFormattedTextClick(e: React.MouseEvent, currentRelativeTeamUrl = '') {
     const hashtagAttribute = (e.target as any).getAttributeNode('data-hashtag');
     const linkAttribute = (e.target as any).getAttributeNode('data-link');
-    const channelMentionAttribute = (e.target as any).getAttributeNode('data-channel-mention');
+    const channelMentionAttribute = (e.currentTarget as any).getAttributeNode('data-channel-mention');
```

Comment:
My bad, I am wrong, i think we just need to use e.target only and then in the same function below we have else if condition 
`else if (channelMentionAttribute) ` we just need to change this to `else if (!channelMentionAttribute)`. Am I on the right track?

Reply:
Oh, I didn't realize this click handler was part of the problem here because it always sends you to the current team.

I don't think we want to change that part actually since we still want to check `if (channelMentionAttribute)`. If we don't, then it'll cause the page to refresh when clicking on a channel link. When we use `e.preventDefault()` and `getHistory().push(...)`, we go to the URL without causing a refresh.

Instead, I think what we want to add a second `data-channel-mention-team` attribute to the link with the team name so that we can use that with `data-channel-mention` to link to the right team. The code that adds that is in `utils/text_formatting.tsx`

Reply:
Weird. https://github.com/mattermost/mattermost/pull/22704 added a `data-channel-mention-team` that's set in some cases, but it's never actually used by `handleFormattedTextClick`, so that doesn't do anything yet 😅

Reply:
😅I actually created a variable with data-channel-team-mention's value but as the code was working even without using it removed it.